c2corg_ui/externs/appx.js

```diff
@@ -84,3 +84,207 @@ appx.BootstrapSliderValues;
+ * }}
+ */
+appx.DocumentAssociations;
+
+
+/**
+ * @typedef {{
+ *   associations: appx.DocumentAssociations,
+ *   locales: Array.<Object>,
+ *   type : string,
+ *   quality: string,
+ *   document_id: number
+ * }}
+ */
+appx.Document;
+
+
+/**
+ * @typedef {{
+ *   associations: appx.DocumentAssociations,
+ *   locales: Array.<Object>,
+ *   type : string,
+ *   quality: string,
+ *   document_id: number,
+ *   
+ *   geometry: Object,
+ *   date_start: (string | Date),
+ *   date_end: (string | Date),
+ *   activities : Array.<Object>,
+ *   frequentation: string,
+ *   participant_count : (number | string),
+ *   elevation_min : number,
+ *   elevation_max : number,
+ *   elevation_up_snow: number,
+ *   elevation_down_snow: number,
+ *   snow_quantity: string,
+ *   snow_quality: string,
+ *   glacier_rating: string,
+ *   duration: number,
+ *   avalanche_signs: string,
+ *   elevation_access: number,
+ *   height_diff_up: number,
+ *   height_diff_down: number,
+ *   length_total: number,
+ *   partial_trip: boolean,
+ *   public_transport: string,
+ *   access_condition: string,
+ *   lift_status: string,
+ *   awesomeness: string,
+ *   condition_rating: string,
+ *   hut_status: string,
+ * }}
+ */
+appx.Outing;
+
+
+/**
+ * @typedef {{
+ *   associations: appx.DocumentAssociations,
+ *   locales: Array.<Object>,
+ *   type : string,
+ *   quality: string,
+ *   document_id: number,
+ *   
+ *   height_diff_up: number,
+ *   height_diff_down: number,
+ *   height_diff_access: number,
+ *   height_diff_difficulties: number,
+ *   lift_access: boolean,
+ *   elevation_min : number,
+ *   elevation_max : number,
+ *   route_type: Array.<string>,
+ *   orientations: Array.<string>,
+ *   durations: number,
+ *   main_waypoint_id: number,
+ *   activities: Array.<Object>,
+ *   route_length: number,
+ *   configuration: Array.<string>,
+ *   ski_rating: string,
+ *   ski_exposition: string,
+ *   labande_ski_rating: string,
+ *   labande_global_rating: string,
+ *   difficulties_height: number,
+ *   engagement_rating: string,
+ *   equipment_rating: string,
+ *   global_rating: string,
+ *   ice_rating: string,
+ *   geometry: Object,
+ *   risk_rating: string,
+ *   rock_types: Array.<string>,
+ *   mixed_rating: string,
+ *   glacier_gear: string,
+ *   exposition_rock_rating: string,
+ *   rock_free_rating: string,
+ *   rock_required_rating: string,
+ *   aid_rating: string,
+ *   climbing_outdoor_type: string,
+ *   hiking_mtb_exposition: string,
+ *   hiking_rating: string,
+ *   snowshoe_rating: string,
+ *   mtb_up_rating: string,
+ *   mtb_down_rating: string,
+ *   mtb_length_asphalt: string,
+ *   mtb_length_trail: string,
+ *   mtb_height_diff_portages: string,
+ *   via_ferrata_rating: string
+ * }}
+ */
+appx.Route;
+
+/**
+ * @typedef {{
+ *   associations: appx.DocumentAssociations,
+ *   locales: Array.<Object>,
+ *   type : string,
+ *   quality: string,
+ *   document_id: number,
+ *   
```

Comment:
I have put this blank line in every document to show that those properties above are the same as found in appx.document.
At first I wanted to create an appx.document and appx.waypoint, so that waypoint inherits from document, but after discussing with @gberaudo it doesn't seem possible, so he suggested that I copy those lines from appx.document and separate them with a blank line.